### ISSUE TITLE
Difference with https://github.com/freeipa/ansible-freeipa

### ISSUE THREAD

**Issue by redhatrises** [RedHatGov/ansible-idm#1]
What's the difference between this repo and ansible content and https://github.com/freeipa/ansible-freeipa? Seems like a lot of duplicative work that is already done?

**Comment by shawndwells** [comment]
Gentle bump (+cc @kjw3). This came up in a conversation today and I wasn't sure how to answer. What are a few lines to describe when to send customers here vs the freeipa ansible stuff?

Thanks!

**Comment by kjw3** [comment]
@redhatrises This is carry over of a role I was holding onto from the original project hattrick repo at https://github.com/redhat-kejones/hattrick. We are currently in the process of refactoring all of that to decouple things and part of this will be duplicate reduction. I'll take a look at the ansible-freeipa repo and see if it's something we can utilize over what we have in this role.

**Comment by kjw3** [comment]
@redhatrises I got a chance to look at this today. I definitely agree that it looks like a lot of the subset of tasking this role does vs the freeipa role(s) is duplicated. However, as of right now, it appears you have to either git clone the freeipa/ansible_freeipa repo directly or install and use mazer to utilize these roles.

Is there a way to get them from galaxy without using installing and using mazer? I have yet to figure out a way.

**Comment by redhatrises** [comment]
> @redhatrises I got a chance to look at this today. I definitely agree that it looks like a lot of the subset of tasking this role does vs the freeipa role(s) is duplicated. However, as of right now, it appears you have to either git clone the freeipa/ansible_freeipa repo directly or install and use mazer to utilize these roles.
> 
> Is there a way to get them from galaxy without using installing and using mazer? I have yet to figure out a way.

@kjw3 curious what other ways you are referring to? If using `ansible-galaxy`:

```
ansible-galaxy install freeipa.ipaserver
ansible-galaxy install freeipa.ipareplica
ansible-galaxy install freeipa.ipaclient
```

**Comment by kjw3** [comment]
yep, that's what I was looking for. It wasn't obvious to me that each role could be pulled individually. Thanks. yesterday I ended up just pointing the src in requirements file to the git repo, but I like pulling the roles individually.

I'll continue investigation and if I find these can be used, will deprecate this role in favor of the freeipa roles.

Thanks for pointing it out.

**Comment by redhatrises** [comment]
> yep, that's what I was looking for. It wasn't obvious to me that each role could be pulled individually. Thanks. yesterday I ended up just pointing the src in requirements file to the git repo, but I like pulling the roles individually.
> 
> I'll continue investigation and if I find these can be used, will deprecate this role in favor of the freeipa roles.
> 
> Thanks for pointing it out.

@kjw3 yeah no problem. Just didn't want us to have to do extra work if we don't have to.

**Comment by nicholasburr** [comment]
@kjw3 Sorry to necro this thread, but has this effort been migrated to freeipa/ansible-freeipa?

**Comment by kjw3** [comment]
> @kjw3 Sorry to necro this thread, but has this effort been migrated to freeipa/ansible-freeipa?

@nicholasburr I would say try the freeipa collection. It definitely has momentum and is more robust. We are still carrying this role because it serves a simple purpose and we just haven't had time yet to migrate everything to use the freeipa roles.
